packages/flutter_tools/lib/src/asset.dart

```diff
@@ -268,20 +269,21 @@ Map<String, dynamic> _readMaterialFontsManifest() {
   final String fontsPath = fs.path.join(fs.path.absolute(Cache.flutterRoot),
       'packages', 'flutter_tools', 'schema', 'material_fonts.yaml');
 
-  return loadYaml(fs.file(fontsPath).readAsStringSync());
+  return castStringKeyedMap(loadYaml(fs.file(fontsPath).readAsStringSync()));
 }
 
 final Map<String, dynamic> _materialFontsManifest = _readMaterialFontsManifest();
 
 List<Map<String, dynamic>> _getMaterialFonts(String fontSet) {
-  return _materialFontsManifest[fontSet];
+  final List<dynamic> fontsList = _materialFontsManifest[fontSet];
+  return fontsList?.map(castStringKeyedMap)?.toList();
```

Comment:
`List.map` should be given a type parameter per [Flutter style guide](https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#avoid-using-var):
```dart
return fontsList?.map<Map<String, dynamic>>(castStringKeyedMap)?.toList();
```

Reply:
done